### ISSUE TITLE
[feat] Code Signing macOS Application in development mode

### ISSUE THREAD

**Issue by wyhaya** [tauri-apps/tauri#7930]
### Describe the problem

In `development mode`, Tauri does not code-sign macOS app. If your app needs to call some API that require `permission` control, you will need to re-authorize each time.

Example:

![preview](https://github.com/tauri-apps/tauri/assets/23690145/3b7b1516-473b-453a-acca-fe830d5fdae3)

If your application saves 100 passwords in the `keychain`, every time you run `tauri dev`, macOS will think it is a new unauthorized app, so you will have to enter the password 100 times in this dialog, which greatly affects the development experience.

---

Code signing in development mode will especially make sense if your app needs to access any of the ones listed below.

![2](https://github.com/tauri-apps/tauri/assets/23690145/cee8e044-bdf0-49a6-a36f-96eef48d043c)

### Describe the solution you'd like

When run `tauri dev`, check the environment variables listed here: [Code Signing macOS Applications](https://tauri.app/v1/guides/distribution/sign-macos#tldr).
If configured, sign the app.

**Comment by thewh1teagle** [comment]
@wyhaya 
Can you tell what's the solution? Or even a workaround?
Should I sign the app with self signed cert everytime it reloads?
Is there a simple solution it will work with `cargo tauri dev`?

**Comment by thewh1teagle** [comment]
After many many solutions I tried including AID repair to the disk, and playing with different options in keychain I found something that worked and stopped this password prompt without errors:
1. Open keychain app from launchpad
2. Find the key (probably with the name of your app) in Login / Local Items
3. Right click on it and choose copy
4. Navigate to system keys, right click and paste the key
5. Delete the key from login
6. Double click the key in system 
7. Switch to access control and choose 'Allow all applications to access this key'
8. Save changes
9. Restart the app and no more password prompts.

TLDR: open keychain and copy your app key name from login / local items into system by right click the key, change tab, and paste. and then delete the old key from login tab. search term: webcrypto

**Comment by rtmotiondev** [comment]
@thewh1teagle Unfortunately that doesn't seem to work for me.

**Comment by thewh1teagle** [comment]
> Unfortunately that doesn't seem to work for me.

Double check it... I encountered the same issue again and that fixed it. macOS m1 14.5

**Comment by RandomEngy** [comment]
I copied the keys to System and they didn't show up at first. It required a restart and I could see them. I changed to "Allow all applications to access this key", but it did not appear that my app could see them, and it created new entries in the login keychain.

I tried changing the existing keys in the login keychain to "Allow all applications" but that didn't work either: it still opened the prompt. On m2 15.1.1.

**Comment by launchthatbrand** [comment]
same issue here. Any fix?

**Comment by RandomEngy** [comment]
I ended up writing some special code that can copy the credentials to insecure storage when in dev mode. You get the prompt once to read from secure storage, but then on subsequent launches it's reading from the insecure copy.

In production it always reads/writes to secure storage.

**Comment by theoephraim** [comment]
Not exactly the same issue but maybe related.

I'm trying to create a keypair in the keychain (secure enclave protected) and running into the dreaded `Code=-34018` which I think means the entitlements are not set correctly. If I understand correctly, these keychain APIs are only accessible when the code is signed, so how is this supposed to work during development if code signing is not enabled?

Any guidance is much appreciated!

**Comment by DamianPereira** [comment]
Is there any progress on this? I'm coding an app that needs reading from the keychain and every re-compile it asks for permission. 

Update, managed to work around this with:

```
// src-tauri/tauri.macos.conf.json
{
  "build": {
    "runner": "../scripts/dev-runner.sh"
  }
}

// dev-script.sh
#!/bin/bash

cargo build "${@:2}" || exit $?

codesign -fs "My signing identity name" target/debug/my-app-name

exec target/debug/my-app-nmae
```

**Comment by wyhaya** [comment]
I solved the annoying pop-ups by writing extra code to store the key locally, which also helped with debugging.


```rs
// Example
#[cfg(all(debug_assertions, target_os = "macos"))]
impl MacDebugEntry {
    pub fn new(user: String) -> Result<Self, std::io::Error> {
        let _ = std::fs::create_dir(".keychain");
        Ok(Self { user })
    }
    fn user(&self) -> String {
        format!(".keychain/{}", self.user)
    }
    pub fn set_password(&self, password: &str) -> Result<(), std::io::Error> {
        std::fs::write(self.user(), password)
    }
    pub fn get_password(&self) -> Result<String, std::io::Error> {
        std::fs::read_to_string(self.user())
    }
    pub fn delete_password(&self) -> Result<(), std::io::Error> {
        std::fs::remove_file(self.user())
    }
}
```
